### ISSUE TITLE
LDC's std.math: soft float isn't properly guarded

### ISSUE THREAD

**Issue by denizzzka** [ldc-developers/ldc#3292]
This code (marked by `<---`) breaks ARM soft fload compilation (with `ARM_SoftFloat` version).
I don't understand what is purpose of this code therefore can't fix it.
Original std.math what packed with DMD isn't contains these lines.

```D
    version (ARM_Any)
    {
        static bool hasExceptionTraps_impl() @safe
        {
            auto oldState = getControlState();
            // If exceptions are not supported, we set the bit but read it back as zero
            // https://sourceware.org/ml/libc-ports/2012-06/msg00091.html
            version (LDC)
            {
                version (AArch64) __asm_trusted("msr FPCR, $0", "r", oldState | divByZeroException);
                else              __asm_trusted("vmsr FPSCR, $0", "r", oldState | divByZeroException); // <--- this
            }
            else
                setControlState(oldState | divByZeroException);
            immutable result = (getControlState() & allExceptions) != 0;
            version (LDC)
            {
                version (AArch64) __asm_trusted("msr FPCR, $0", "r", oldState);
                else              __asm_trusted("vmsr FPSCR, $0", "r", oldState); // <--- and this
            }
            else
                setControlState(oldState);
            return result;
        }
    }
```

**Comment by kinke** [comment]
I had a go at this a while back, but lost interest: https://github.com/ldc-developers/phobos/pull/66

**Comment by kinke** [comment]
Fixed upstream and cherry-picked in #3303.
